### DISCUSSION TITLE
SendHttpRequest get the results

### DISCUSSION THREAD

**Discussion by HolgerHuckfeldt** [elsa-workflows/elsa-core#1318] [Q&A] [answered]
I'm trying to use the SendHttpRequest activity to call a rest-api to get some data. In my example, the activities name is "Call" and I call this REST-Service: https://api.predic8.de/shop/products/62

It's returning this json:
{  "name": "Mango fresh", "price": 5.55, "photo_url": "/shop/products/62/photo",  "category_url": "/shop/categories/Fresh"}

I cant get the resulting json in the following actions. I hoped it is like in an endpoint, the result should be in the output. But {{ Input }} in the following activity is empty, so is {{ Activities.Call.Output }} 

Of course "Read Content" is checked. What am I missing?

I use Version 2.1.1.463

Thanks.

**Comment by sfmskywalker** [comment, marked as answer]
Sorry about that, the output property of the `SendHttpRequest` activity was recently changed. It now has two output properties:

- `Response: HttpResponseModel`
- `ResponseContent: object?`

The body of the response of your request will be stored in the `ResponseContent` property, which you can access like this:

```handlebars
Product name: {{ Activities.Call.ResponseContent.name }}
```

Notice that in the above snippet, I am accessing the `name` field of the JSON response returned by the [API call](https://api.predic8.de/shop/products/62).

**Reply by HolgerHuckfeldt** [reply]
Great, that works.
It is a bit confusing that the `HttpEndpoint` and `HttpRequest` share there data in a different way. Are there plans to change the `HttpEndpoint` too?

**Reply by swuecho** [reply]
> Great, that works.
> It is a bit confusing that the `HttpEndpoint` and `HttpRequest` share there data in a different way. Are there plans to change the `HttpEndpoint` too?

Could Your provide an example or write up a tutorial. Can not get a simple workflow to work.

Assuming, set a request to https://restcountries.eu/rest/v2/all every 10s and write the response to console. 

I could get the cron and send request work, but have difficulties to print to console.

PS: I read some very interesting guides. However, I think short and simple one are more helpful to learn.

**Reply by sfmskywalker** [reply]
I attached a sample workflow you can import into your server via the dashboard app.

[scraper-workflow.zip](https://github.com/elsa-workflows/elsa-core/files/7117298/scraper-workflow.zip)

Here's what it does:

1. Every 5 seconds,
2. Make a call to https://restcountries.eu/rest/v2/all
3. For each country in the returned list,
4. Write the name of the country

Sample output I see every 5 seconds in the console:

![image](https://user-images.githubusercontent.com/938393/132249012-a284f238-2d13-4e52-9b68-3e979c8e144f.png)

**Reply by swuecho** [reply]
Thanks. I missed the '()' after 'ResponseConent'. 

```js
activities.nameOfActivity.ResponseContent()
```

I check the http code 

```c#
        [ActivityOutput] public HttpResponseModel? Response { get; set; }
        [ActivityOutput] public object? ResponseContent { get; set; }
```

why `ResponseContent` become a method in javascript?

PS: find the doc https://elsa-workflows.github.io/elsa-core/docs/expressions/expressions-javascript#activity-output-elsa-21
